src/marten/db/query/sql/query.cr

```diff
@@ -313,6 +308,33 @@ module Marten
             {sql, parameters}
           end
 
+          private def build_count_column_specific_subquery_query(s, column_name : String | Symbol)
+            column_name = column_name.to_s
+
+            s << "SELECT COUNT(#{column_name})"
+            s << "FROM ("
+            s << "SELECT"
+
+            if distinct
+              s << connection.distinct_clause_for(distinct_columns)
+            end
+
+            s << "#{Model.db_table}.#{column_name}"
```

Comment:
In order to simplify you could make use of `#build_sql` in this method to replace the use of the `s` argument. That way you would return the subquery string directly:

```crystal
column_name = column_name.to_s
  
build_sql do |s|
  s << "SELECT COUNT(#{column_name})"
  s << "FROM ("
  s << "SELECT"
  
  if distinct
    s << connection.distinct_clause_for(distinct_columns)
  end
  
  s << "#{Model.db_table}.#{column_name}"
end
```